flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkConfigBuilder.java

```diff
@@ -409,15 +419,44 @@ private static void setPodTemplate(
             effectiveConfig.setString(
                     podConfigOption,
                     createTempFile(
-                            mergePodTemplates(
-                                    basicPod,
-                                    appendPod,
-                                    effectiveConfig.get(
-                                            KubernetesOperatorConfigOptions
-                                                    .POD_TEMPLATE_MERGE_BY_NAME))));
+                            applyResourceToPodTemplate(
+                                    mergePodTemplates(
+                                            basicPod,
+                                            appendPod,
+                                            effectiveConfig.get(
+                                                    KubernetesOperatorConfigOptions
+                                                            .POD_TEMPLATE_MERGE_BY_NAME)),
+                                    resource)));
         }
     }
 
+    private static Pod applyResourceToPodTemplate(Pod podTemplate, Resource resource) {
+        if (resource != null
+                && !StringUtils.isNullOrWhitespaceOnly(resource.getEphemeralStorage())
+                && podTemplate != null
+                && podTemplate.getSpec() != null) {
```

Comment:
This will ignore the ephemeralStorage setting if the user did not specify any podTemplate (or spec in the podTemplate), that is probably incorrect.

Reply:
We need some more test coverage for different configs:
 1. Ephemeral storage settings without any podtemplate 
 2. Ephemeral storage without tm/jm podTemplate 
 3. Ephemeral storage without spec in podTemplate

Reply:
Added different scenarios in FlinkConfigBuilderTest